programs/exchange/src/math.rs

```diff
@@ -639,15 +822,125 @@ mod tests {
                 ..Default::default()
             };
             let fee = 300u32;
-            let result =
-                calculate_swap_out_amount(&assetUSD, &assetBTC, &(50000 * 10u64.pow(6)), &fee);
+            let result = calculate_swap_out_amount(&assetUSD, &assetBTC, 50000 * 10u64.pow(6), fee);
             assert_eq!(result, 0_99700000);
-            let result = calculate_swap_out_amount(&assetBTC, &assetUSD, &(1 * 10u64.pow(8)), &fee);
+            let result = calculate_swap_out_amount(&assetBTC, &assetUSD, 1 * 10u64.pow(8), fee);
             assert_eq!(result, 49850_000_000);
-            let result = calculate_swap_out_amount(&assetBTC, &assetETH, &99700000, &fee);
+            let result = calculate_swap_out_amount(&assetBTC, &assetETH, 99700000, fee);
             assert_eq!(result, 24_850_2250);
         }
     }
+    #[test]
+    fn test_calculate_burned_shares() {
+        {
+            // 7772,102...
+            let asset = Asset {
+                price: 14 * 10u64.pow(PRICE_OFFSET.into()),
+                supply: 100 * 10u64.pow(6),
+                last_update: 10,
+                decimals: 6,
+                ..Default::default()
+            };
+            let user_debt = 1598;
+            let user_shares = 90;
+            let amount = 9857;
+            let burned_shares = calculate_burned_shares(&asset, user_debt, user_shares, amount);
```

Comment:
I think we should refactor this function to use all_shares and all_debt instead of user specific.